commander3/src/comm_param_mod.f90

```diff
@@ -235,52 +235,45 @@ subroutine read_comm_params(cpar)
     type(hash_tbl_sll) :: htable
     type(comm_params), intent(inout) :: cpar
     
-    integer(i4b)       :: paramfile_len, ierr, i
-    character(len=512) :: paramfile
+    integer(i4b)       :: paramfile_len, ierr, i, idx
+    character(len=512) :: paramfile, paramfile_name
     character(len=512), allocatable, dimension(:) :: paramfile_cache
 
     call getarg(1, paramfile)
     ! read parameter file once, save to ascii array
-    ! Need to know how long the file is to allocate ascii array
+    ! Guess at file size, will be resized later if needed
+
     if (cpar%myid == cpar%root) then
-       call get_file_length(paramfile,paramfile_len)
+      paramfile_len = 512
```

Comment:
Why 512? Most parameter files are quite a bit longer than this. How about 5000?

Reply:
This was a test to make sure that the code that auto-updates the length worked properly. I agree that in general a longer one would be slightly more optimal here.